ironfish/src/rpc/routes/accounts/getTransaction.ts

```diff
@@ -80,12 +80,12 @@ router.register<typeof GetAccountTransactionRequestSchema, GetAccountTransaction
     const serializedNotes = notes.map(serializeRpcAccountDecryptedNote)
     const serializedTransaction = serializeRpcAccountTransaction(transaction)
 
-    const status = await getTransactionStatus(
-      node,
-      transaction.blockHash,
-      transaction.sequence,
-      transaction.transaction.expirationSequence(),
-    )
+    const headSequence = await node.accounts.getAccountHeadSequence(account)
+    const minimumBlockConfirmations = node.config.get('minimumBlockConfirmations')
+
+    const status = headSequence
+      ? account.getTransactionStatus(transaction, headSequence, minimumBlockConfirmations)
```

Comment:
I have the feeling this code would be a little cleaner if it was on the wallet and not on an account. Then the wallet already knows about all the pre-requisite data.